openedx/core/djangoapps/credentials/management/commands/notify_credentials.py

```diff
@@ -45,6 +46,25 @@ def parsetime(timestr):
     return dt
 
 
+def paged_query(queryset, delay, page_size):
+    """
+    A generator that iterates through a queryset but only resolves chunks of it at once, to avoid overwhelming memory
+    with a giant query. Also adds an optional delay between yields, to help with load.
+    """
+    count = queryset.count()
+    pages = int(math.ceil(count / page_size))
+
+    for page in range(pages):
+        page_start = page * page_size
+        page_end = page_start + page_size
+        subquery = queryset[page_start:page_end]
```

Comment:
I might be reading this wrong, but if count isn't evenly divisible by page_size, wouldn't you reach a point where page_end is past the end of the queryset?